OmiseSwift/Balance.swift

```diff
@@ -13,9 +13,9 @@ public class Balance: ResourceObject {
         set { set("total", Int64Converter.self, toValue: newValue) }
     }
     
-    public var currency: String? {
-        get { return get("currency", StringConverter.self) }
-        set { set("currency", StringConverter.self, toValue: newValue) }
+    public var currency: Currency? {
+        get { return get("currency", StringConverter.self).flatMap(Currency.init(code:)) }
```

Comment:
We can't make `CurrencyConverter` for now. Converter requires that its `Target` type must be a subclass of `NSObject`. Currency is an associated value enum so we're out of luck for now.
I already changed the Converter's Target type to `Any` in the swift-3 branch which should allow us to do `CurrencyConverter`